### PR TITLE
[ActionSheet] Add test for color themer

### PR DESCRIPTION
**Pull request by codeman7**
Add test to make sure color themer works as intended.
Depends on #4984 & #4968

### CODE REVIEW

File: components/ActionSheet/tests/unit/ActionSheetTest.swift [hunk 1]
```diff
@@ -78,4 +81,72 @@ class ActionSheetTest: XCTestCase {
...
+  func testColorThemer() {
+    // Given
+    let titleTest = "Title"
+
+    // When
+    actionSheet.title = titleTest
+
+    // Then
+    if let title = getLabel(with: titleTest) {
+      XCTAssertEqual(title.textColor, .black)
```
Comment (romoore) [inline components/ActionSheet/tests/unit/ActionSheetTest.swift:94]: This doesn't seem related to the "when" condition. Why would changing the title change the color of the text?

File: components/ActionSheet/tests/unit/ActionSheetTest.swift [hunk 2]
```diff
@@ -78,4 +81,72 @@ class ActionSheetTest: XCTestCase {
...
+    } else {
+      XCTFail("No title label")
+    }
+
+    // Given
```
Comment (romoore) [inline components/ActionSheet/tests/unit/ActionSheetTest.swift:99]: Please split these into separate tests.

**Review by romoore — DISMISSED** [review]

File: components/ActionSheet/tests/unit/ActionSheetTest.swift (hunk 1 above)
Comment (codeman7) [reply components/ActionSheet/tests/unit/ActionSheetTest.swift:94]: Make sure the default behavior is as intended. Split into a separate test.

File: components/ActionSheet/tests/unit/ActionSheetTest.swift (hunk 2 above)
Comment (codeman7) [reply components/ActionSheet/tests/unit/ActionSheetTest.swift:99]: Done

File: components/ActionSheet/tests/unit/ActionSheetTest.swift [hunk 3]
```diff
@@ -78,4 +81,77 @@ class ActionSheetTest: XCTestCase {
...
+    if let title = getLabel(with: titleTest), let message = getLabel(with: messageTest) {
+      XCTAssertEqual(title.textColor, colorScheme.onSurfaceColor.withAlphaComponent(0.87))
+      XCTAssertEqual(message.textColor, colorScheme.onSurfaceColor.withAlphaComponent(0.6))
+    } else {
+      XCTFail("One of the labels wasn't set correctly")
+    }
+
+  }
+
+  func getLabel(with text: String) -> UILabel? {
```
Comment (yarneo) [inline components/ActionSheet/tests/unit/ActionSheetTest.swift:143]: why is it so difficult to get this label? What is this label for the ActionSheet?

Comment (codeman7) [reply components/ActionSheet/tests/unit/ActionSheetTest.swift:143]: The header is an `ivar` of the view controller and the labels are `ivar` of the header.

Comment (romoore) [reply components/ActionSheet/tests/unit/ActionSheetTest.swift:143]: Please consider declaring them as properties in a class continuation category so they can be re-declared in a Testing category here.

File: components/ActionSheet/tests/unit/ActionSheetTest.swift [hunk 4]
```diff
@@ -78,4 +81,77 @@ class ActionSheetTest: XCTestCase {
...
+
+    // Given
+    actionSheet.message = messageTest
+    if let title = getLabel(with: titleTest), let message = getLabel(with: messageTest) {
+      XCTAssertEqual(title.textColor, colorScheme.onSurfaceColor.withAlphaComponent(0.87))
+      XCTAssertEqual(message.textColor, colorScheme.onSurfaceColor.withAlphaComponent(0.6))
+    } else {
+      XCTFail("One of the labels wasn't set correctly")
+    }
+
```
Comment (romoore) [inline components/ActionSheet/tests/unit/ActionSheetTest.swift:140]: Nit: Extra newline.

File: components/ActionSheet/tests/unit/ActionSheetTest.swift (hunk 2 above)
Comment (romoore) [reply components/ActionSheet/tests/unit/ActionSheetTest.swift:99]: Looks like there are still two different tests in here. If these should be the same test, please combine the related given, when, and then clauses.

File: components/ActionSheet/tests/unit/ActionSheetTest.swift [hunk 5]
```diff
@@ -78,4 +81,77 @@ class ActionSheetTest: XCTestCase {
...
+    MDCActionSheetColorThemer.applySemanticColorScheme(colorScheme, to: actionSheet)
+
+    // Then
+    if let title = getLabel(with: titleTest) {
+      XCTAssertEqual(title.textColor, colorScheme.onSurfaceColor.withAlphaComponent(0.6))
+    } else {
+      XCTFail("No title label")
+    }
+
+    // Given
```
Comment (romoore) [inline components/ActionSheet/tests/unit/ActionSheetTest.swift:132]: Also looks like a second test is starting here.  Please split these up or combine their sections.

File: components/ActionSheet/tests/unit/ActionSheetTest.swift [hunk 6]
```diff
@@ -78,4 +81,77 @@ class ActionSheetTest: XCTestCase {
...
+    let messageTest = "Message"
+
+    // When
+    actionSheet.message = messageTest
+
+    // Then
+    if let message = getLabel(with: messageTest) {
+      XCTAssertEqual(message.textColor, .black)
```
Comment (romoore) [inline components/ActionSheet/tests/unit/ActionSheetTest.swift:107]: This seems to be missing an assert for the title text color once a `message` value is assigned.

**Review by romoore — CHANGES_REQUESTED** [review]
Based on comments in #4984, this is supposed to be all of the unit tests for that PR.  Please be sure to test the remaining APIs being added:

* titleTextColor
* messageTextColor
* actionTextColor
* actionTintColor

File: components/ActionSheet/tests/unit/ActionSheetTest.swift (hunk 2 above)
Comment (codeman7) [reply components/ActionSheet/tests/unit/ActionSheetTest.swift:99]: Done

File: components/ActionSheet/tests/unit/ActionSheetTest.swift (hunk 4 above)
Comment (codeman7) [reply components/ActionSheet/tests/unit/ActionSheetTest.swift:140]: Done

**Comment by codeman7** [comment]
Please don't review until #4984 has landed.
